system/jlib/jsmartsock.ipp

```diff
@@ -117,5 +118,31 @@ private:
     StringAttr msg;
 };
 
+class jlib_decl CSmartSocket: implements ISmartSocket, public CInterface
+{
+
+public:
+    IMPLEMENT_IINTERFACE;
+
+    ISocket *sock;
+    SocketEndpoint ep;
+    CSmartSocketFactory *factory;
+
+    CSmartSocket(ISocket *_sock, SocketEndpoint &_ep, CSmartSocketFactory *_factory);
```

Comment:
Not new, but could pass in ISmartSocketFactory instead of implementation which would be a bit cleaner.